imagebuilder_check/check.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python3
+"""
+Reads and analyses the logs for imagebuilder
+"""
+from datetime import datetime
+import os
+import sys
+import json
+
+NAGIOS_STATE_OK             = 0
+NAGIOS_STATE_WARNING        = 1
+NAGIOS_STATE_CRITICAL       = 2
+
+WAITED_FOR_TOO_LONG = 90000 # 25 hours
+
+
+
+def get_run_data(filename: str, cloud: str) -> dict:
+    """
+    Loads filename and finds the last run of cloud
+
+    Returns
+    -------
+    dict
+        Dictionary containing the data of the last run
+    """
+    json_data = None
+    try:
+        with open(filename, "r", encoding="utf-8") as f:
+            json_data = json.load(f)
+    except IOError as e:
+        print(f"Failed to open file: {e}")
+        sys.exit(NAGIOS_STATE_CRITICAL)
+
+    for run in reversed(json_data):
+        if run["cloud"] == cloud:
+            return run
+
+    print("No runs in the log files!")
+    sys.exit(NAGIOS_STATE_CRITICAL)
+
+
+
+
+
+def main() -> None:
+    """
+    Main entry point
+    """
+
+    cloud = os.getenv("IMAGEBUILDER_CHECK_CLOUD")
+    filename = os.getenv("IMAGEBUILDER_CHECK_FILE")
+    input_json = os.getenv("IMAGEBUILDER_INPUT_FILE")
+
+    if cloud is None or filename is None:
+
+        missing = [
+            x for x in
+                ["IMAGEBUILDER_CHECK_CLOUD", "IMAGEBUILDER_CHECK_FILE", "IMAGEBUILDER_INPUT_FILE"]
+            if os.getenv(x) is None
+            ]
+
+        raise EnvironmentError(f"Environment variables are not set! ({', '.join(missing)})")
+
+
+    input_json_data = None
+    with open(input_json, "r", encoding="utf-8") as f:
+        input_json_data = json.load(f)
+
+    if input_json_data is None:
+        print(f"Failed to read json file: {input_json}")
+        sys.exit(NAGIOS_STATE_CRITICAL)
+
+
+    run_data = get_run_data(filename, cloud)
+
+
+    if (
+        datetime.now() - datetime.strptime(run_data["start_timestamp"], "%Y-%m-%d %H:%M:%S.%f")
+        ).seconds > WAITED_FOR_TOO_LONG:
+
+        print("Imagebuilder has not been run in a while!")
```

Comment:
`Imagebuilder has not been run in a while!` <- maybe it would be a bit more informative something like `Imagebuilder was run last time more than X hours ago`